at_client/src/main/java/org/atsign/client/cli/Register.java

```diff
@@ -1,102 +1,167 @@
 package org.atsign.client.cli;
 
+import java.util.Map;
 import java.util.Scanner;
+import java.util.concurrent.Callable;
 
 import org.atsign.client.util.RegisterUtil;
 import org.atsign.common.AtSign;
 import org.atsign.config.ConfigReader;
 
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+
 /**
  * Command line interface to claim a free atsign. Requires one-time-password
  * received on the provided email to validate.
  * Registers the free atsign to provided email
  */
-public class Register {
-    public static void main(String[] args) throws Exception {
-        if (args.length != 1) {
+@Command(name = "register", description = "Get an atsign and register")
+public class Register implements Callable<String> {
```

Comment:
@srieteja can you refactor in the following lines:

Let's split the code into smaller unit testable methods.

Code has the following logical parts:

1. Decide which API to invoke
2. Invoke the API
	- Success path
	- Failure path
3. Enter OTP
	- Success path
	- Failure path (Retry comes here)
4. Activate	
	- Success path
	- Failure path



	main() {
		String endpoint = RegisterAPI.getEndpoint(<params>);
		RegisterAPI.invoke(endpoint, handler);
                 // Rest of the logic
	}

	RegisterAPI {
		static String getEndpoint(<params>)  {

		}

		void invoke(String endPoint, InvocationHandler handler) {

				// Invoke endPoint
				// If success call handler.successCallback();
				// If failure call handler.failureCallback();
				// if(retry) {
					invoke();
				}

		}
	}

	InvocationHandler {
		abstract void successCallback();
		abstract void failureCallback();
		abstract boolean retry();
	}

Reply:
Abstractions are suggestive. Please analyze and come up with your version.